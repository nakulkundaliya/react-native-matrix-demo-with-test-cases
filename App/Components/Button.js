import React, { Component } from 'react';
// import PropTypes from 'prop-types';
import { TouchableOpacity, Text } from 'react-native';
import styles from './Styles/ButtonStyle';

const Button = props => {
  const { name, onPress, style, disabled } = props;
  return (
    <TouchableOpacity
      style={[styles.buttonStyle, style]}
      onPress={() => onPress()}
      disabled={disabled}
    >
      <Text style={styles.buttonTextStyle}>{name}</Text>
    </TouchableOpacity>
  );
};
export default Button;
