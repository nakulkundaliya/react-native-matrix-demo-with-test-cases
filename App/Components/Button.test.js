import React from 'react';
import { shallow } from 'enzyme';
import { TouchableOpacity, Text } from 'react-native';

import Button from './Button';

describe('Button', () => {
  describe('Rendering', () => {
    let component;
    const mockFn = jest.fn();

    beforeEach(() => {
      component = shallow(<Button name="test" onPress={mockFn} />);
    });

    it('should be defined', () => {
      expect(Button).toBeDefined();
    });

    it('should match to snapshot', () => {
      expect(component).toMatchSnapshot();
    });

    it('should match button text', () => {
      expect(component.find(Text).props('children').children).toEqual('test');
    });

    it('should click button', () => {
      component.find(TouchableOpacity).simulate('press');
      expect(mockFn).toHaveBeenCalled();
    });
  });
});
