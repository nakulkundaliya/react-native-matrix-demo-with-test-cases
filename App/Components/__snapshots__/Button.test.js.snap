// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Rendering should match to snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    name="test"
    onPress={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "activeOpacity": 0.2,
      "children": <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "white",
          }
        }
      >
        test
      </Text>,
      "disabled": undefined,
      "onPress": [Function],
      "style": Array [
        Object {
          "backgroundColor": "#6e7cea",
          "borderRadius": 10,
          "height": 50,
          "justifyContent": "center",
          "marginVertical": 20,
          "width": 250,
        },
        undefined,
      ],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "test",
        "style": Object {
          "alignSelf": "center",
          "color": "white",
        },
      },
      "ref": null,
      "rendered": "test",
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "activeOpacity": 0.2,
        "children": <Text
          style={
            Object {
              "alignSelf": "center",
              "color": "white",
            }
          }
        >
          test
        </Text>,
        "disabled": undefined,
        "onPress": [Function],
        "style": Array [
          Object {
            "backgroundColor": "#6e7cea",
            "borderRadius": 10,
            "height": 50,
            "justifyContent": "center",
            "marginVertical": 20,
            "width": 250,
          },
          undefined,
        ],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "test",
          "style": Object {
            "alignSelf": "center",
            "color": "white",
          },
        },
        "ref": null,
        "rendered": "test",
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
